go18_test.go

```diff
@@ -317,3 +317,22 @@ func TestTxOptions(t *testing.T) {
 		t.Errorf("Expected error to mention isolation level, got %q", err)
 	}
 }
+
+func TestPing(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	db := openTestConn(t)
+	defer db.Close()
+
+	if err := db.PingContext(ctx); err != nil {
+		t.Fatal("expected success")
+	}
+
+	cancel()
+
+	if err := db.PingContext(ctx); err == nil {
```

Comment:
Can we come up with more tests that don't use context cancellation? I'd like a test where the conn has actually died and the query itself fails.

Reply:
Is there a method in Postgres via Travis that you're aware of to temporarily prevent access?

Alternatively, could grab one of the test SQL drivers that returns a good conn, and then mock the Ping call to return an error. It looks though like the driver is trying to stay using actual Postgres as much as possible.

Reply:
Here's a potential solution, what do you think..

Start a transaction, execute a query to grab the pid that postgres has given that open connection.

Grab a new connection, and execute `SELECT pg_terminate_backend(pid)` from the in-progress transaction to kill the backing process. This should cause the connection to die, and the commit should fail.